test/lit/passes/code-folding_branch-hints.wast

```diff
@@ -0,0 +1,96 @@
+;; NOTE: Assertions have been generated by update_lit_checks.py --all-items and should not be edited.
+;; NOTE: This test was ported using port_passes_tests_to_lit.py and could be cleaned up.
+
+;; RUN: wasm-opt %s -all --code-folding -S -o - | filecheck %s
+
+(module
+ ;; CHECK:      (type $0 (func (param i32 i32) (result f32)))
+
+ ;; CHECK:      (func $different (type $0) (param $x i32) (param $y i32) (result f32)
+ ;; CHECK-NEXT:  (drop
+ ;; CHECK-NEXT:   (local.get $x)
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT:  (@metadata.code.branch_hint "\00")
+ ;; CHECK-NEXT:  (if
+ ;; CHECK-NEXT:   (local.get $x)
+ ;; CHECK-NEXT:   (then
+ ;; CHECK-NEXT:    (nop)
+ ;; CHECK-NEXT:   )
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT:  (f32.const 0)
+ ;; CHECK-NEXT: )
+ (func $different (param $x i32) (param $y i32) (result f32)
```

Comment:
It might be interesting to have the reversed test as well.

Reply:
Added.